src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -82,6 +82,64 @@ public T get(final int... keys) {
         return (T)myData[keys[numNestedDimensions]];
     }
 
+    @SuppressWarnings("unchecked")
+    private T leaf(final int key, final Object[] array){
+        return array == null ? null : (T) array[key];
```

Comment:
i'd rather not. catching ClassCastException seems nasty. Let's wait until/if someone gets stung by this, ok?